hathor/verification/vertex_verifier.py

```diff
@@ -12,11 +12,53 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from typing import Optional
+
 from hathor.conf.settings import HathorSettings
+from hathor.transaction import BaseTransaction
 
 
 class VertexVerifier:
     __slots__ = ('_settings', )
 
     def __init__(self, *, settings: HathorSettings):
         self._settings = settings
+
+    def verify_parents(self, vertex: BaseTransaction) -> None:
+        """All parents must exist and their timestamps must be smaller than ours.
+
+        Also, txs should have 2 other txs as parents, while blocks should have 2 txs + 1 block.
+
+        Parents must be ordered with blocks first, followed by transactions.
+
+        :raises TimestampError: when our timestamp is less or equal than our parent's timestamp
+        :raises ParentDoesNotExist: when at least one of our parents does not exist
+        :raises IncorrectParents: when tx does not confirm the correct number/type of parent txs
+        """
+        vertex.verify_parents()
+
+    @classmethod
```

Comment:
Why is it a class method?

Reply:
Changed in https://github.com/HathorNetwork/hathor-core/pull/797/commits/105e0598363e0125e9f6112ec72b213889790b61

Reply:
Thanks!